Assets/Scripts/InventoryAndItemsScripts/Shop/ShopController.cs

```diff
@@ -0,0 +1,266 @@
+using GameStateManagement;
+using Inventory;
+using Items;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using TMPro;
+using Unity.VisualScripting;
+using UnityEngine;
+using UnityEngine.InputSystem;
+using UnityEngine.InputSystem.Interactions;
+using UnityEngine.UI;
+using UnityEngine.UIElements;
+using static UnityEngine.InputSystem.InputAction;
+
+public class ShopController : MonoBehaviour
+{
+    public List<ItemData> shopItems; // place items to be displayed in shop
+
+    #region Keybinds
+    [SerializeField]
+    [Tooltip("The keybind that controls opening and closing the inventory.")]
+    InputAction InventoryAction;
```

Comment:
from a design perspective, the button that opens the inventory should probably be the standard interact button, and the button that closes the inventory should be the standard back button (for example: on keyboard, e and esc, on controller, a and b)